### ISSUE TITLE
False positive: object-curly-newline with multiline: true

### ISSUE THREAD

**Issue by Eschon** [eslint/eslint#9338]
**Tell us about your environment**

* **ESLint Version:** 4.7.1
* **Node Version:** 8.2.1
* **npm Version:** 5.4.2

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "extends": "airbnb-base",
    "env": {
      "browser": true
    },
    "parserOptions": {
      "ecmaVersion": 2017
    },
    "rules": {
      "brace-style": [2, "stroustrup"],
      "class-methods-use-this": [2, {
        "exceptMethods": [
          "createIconList",
          "createIconSvg",
          "updateIconSize",
          "getDifficultyColor",
          "getSubControl",
          "template",
          "itemTemplate"
        ]
      }],
      "max-len": [1, 100, 2],
      "no-inner-declarations": 0,
      "no-param-reassign": 1,
      "no-plusplus": 0,
      "no-restricted-syntax": [
        "warn",
        "ForInStatement",
        "LabeledStatement",
        "WithStatement"
      ],
      "no-underscore-dangle": 1,
      "no-use-before-define": 0,
      "no-var": 1,
      "object-curly-newline": [2, {
        "ObjectExpression": "always",
        "ObjectPattern": { "multiline": true }
      }],
      "one-var": 0,
      "one-var-declaration-per-line": [2, "initializations"],
      "prefer-const": 0,
      "prefer-destructuring": 0,
      "prefer-template": 1,
      "strict": 0,
      "import/first": 1
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
/* global $ */

$.ajax('/test', { dataType: 'xml' }).done((xml) => {
  console.log(xml);
});
```

**What did you expect to happen?**

No error only the warning about console.log

**What actually happened? Please include the actual, raw output from ESLint.**

I got 2 errors because of `object-curly-newline`

The output:
```
/Users/egon/Desktop/lynx/resources/test.js
  3:17  error    Expected a line break after this opening brace   object-curly-newline
  3:35  error    Expected a line break before this closing brace  object-curly-newline
  4:3   warning  Unexpected console statement                     no-console

✖ 3 problems (2 errors, 1 warning)
  2 errors, 0 warnings potentially fixable with the `--fix` option.
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. It looks like you have the `object-curly-newline` rule configured with the `"ObjectExpression": "always"` option, so the rule is requiring newlines for all object literals. If you only want to require newlines for multiline object literals, you should configure it with "ObjectExpression": "multiline",

**Comment by Eschon** [comment]
I'm sorry you're right. I didn't pay enough attention to the documentation and therefore didn't understand what `objectExpression` and `objectPattern` actually meant.
